test/register.test.js

```diff
@@ -128,31 +128,23 @@ test('awaitable register and after', async t => {
 
 test('awaitable after error handling', async t => {
   const fastify = Fastify()
-  let first = false
-  let second = false
-  let third = false
 
-  // Note that this does not throw in case of error,
-  // all errors are currently collected to ready() because it's
-  // not really possible to do anything about them.
-  await fastify.register(async (instance, opts, next) => {
-    first = true
-    throw new Error('kaboom')
+  const e = new Error('kaboom')
+
+  // TODO this should throw for some reason, and it's not.
+  await fastify.register(async (instance, opts) => {
+    throw e
   })
-  t.is(first, true)
 
-  fastify.register(async (instance, opts, next) => {
-    second = true
+  fastify.register(async (instance, opts) => {
+    t.fail('should not be executed')
```

Comment:
This is executed: the error is bumped to the `ready` but it doesn't stop the loading (adding chaos to an application startup)

Reply:
The test are passing, how can you tell that this is executed?

Reply:
I tried locally and it was not passing (I run `npm update` without cleaning the node module)

Reply:
you need a npm install, it’s a fixed dependency update